icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -769,6 +769,58 @@ MeasureUnit MeasureUnit::product(const MeasureUnit& other, UErrorCode& status) c
     return std::move(impl).build(status);
 }
 
+/**
+ * Searches the `simplifiedUnits` for a unit with the same base identifier in the `newUnit`, for example
+ * `meter` and `meter` or `millimeter` and `centimeter`.
+ * After finding a match, the matched unit in simplified unit will be merged with the `newUnit` and
+ * `true` will be returned. Otherwise, `false` will be returned.
+ */
+bool findAndSet(MaybeStackVector<MeasureUnit> &simplifiedUnits, const MeasureUnit &newUnit,
+                UErrorCode &status) {
+    for (int i = 0, n = simplifiedUnits.length(); i < n; i++) {
+        auto simplifiedUnitImpl = SingleUnitImpl::forMeasureUnit(*simplifiedUnits[i], status);
+        auto newUnitImpl = SingleUnitImpl::forMeasureUnit(newUnit, status);
+        if (simplifiedUnitImpl.identifier == newUnitImpl.identifier) {
+            int32_t newDimensionality = simplifiedUnitImpl.dimensionality + newUnitImpl.dimensionality;
+            UMeasureSIPrefix newSIprefix = static_cast<UMeasureSIPrefix>(
+                simplifiedUnitImpl.siPrefix * simplifiedUnitImpl.dimensionality +
+                newUnitImpl.siPrefix * newUnitImpl.dimensionality);
+
+            auto &simplifiedUnit = *simplifiedUnits[i];
+            simplifiedUnit = simplifiedUnit.withDimensionality(newDimensionality, status);
+            simplifiedUnit = simplifiedUnit.withSIPrefix(newSIprefix ,status);
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+MeasureUnit MeasureUnit::simplify(UErrorCode &status) const {
+    MeasureUnit result;
+    if (this->getComplexity(status) == UMeasureUnitComplexity::UMEASURE_UNIT_MIXED) {
+        status = U_INTERNAL_PROGRAM_ERROR;
+        return result;
+    }
+
+    int32_t outCount;
+    auto singleUnits = this->splitToSingleUnits(outCount, status);
+
+    MaybeStackVector<MeasureUnit> simplifiedUnits;
+    for (int i = 0 ; i < outCount ; ++i) {
+        if (findAndSet(simplifiedUnits,singleUnits[i] , status )) { continue;}
```

Comment:
"findAndSet" is not clear to me.

How about something like: "if (mergeWithExisting(simplifiedUnits, singleUnits[i], status)) { ... }"?

Or combineWithExisting?

Reply:
I made it findAndMerge, what do you think?

Reply:
That helps already. I'll mark as resolved, and see what I feel when I next read this code and docs.